Firestore/core/src/firebase/firestore/remote/grpc_connection.mm

```diff
@@ -181,6 +207,31 @@
   active_calls_.erase(found);
 }
 
+/*static*/ void GrpcConnection::UseTestCertificate(
+    absl::string_view certificate_path, absl::string_view target_name) {
+  HARD_ASSERT(!certificate_path.empty(), "Empty path to test certificate");
+  HARD_ASSERT(!target_name.empty(), "Empty SSL target name");
+
+  if (!test_credentials_) {
+    // Deliberately never deleted.
+    test_credentials_ = new TestCredentials{};
+  }
+
+  test_credentials_->certificate_path = certificate_path.data();
```

Comment:
I don't think this is safe? string_view::data() isn't guaranteed to be null terminated, and I think std::string's operator= assumes the pointer is.

Maybe:
test_credentials_->certificate_path = std::string(cert_path.data(), cert_path.size());

Also immediately below.

Reply:
Done.